backend/FwLite/MiniLcm.Tests/AutoFakerHelpers/EntryFakerHelper.cs

```diff
@@ -0,0 +1,88 @@
+﻿using MiniLcm.Models;
+using Soenneker.Utils.AutoBogus;
+
+namespace MiniLcm.Tests.AutoFakerHelpers;
+
+public static class EntryFakerHelper
+{
+    public static async Task<Entry> EntryReadyForCreation(this AutoFaker autoFaker,
+        IMiniLcmApi api,
+        Guid? entryId = null,
+        bool createComplexForms = true,
+        bool createComplexFormTypes = true,
+        bool createComponents = true)
+    {
+        var entry = autoFaker.Generate<Entry>();
+        if (entryId.HasValue) entry.Id = entryId.Value;
+        if (createComponents) await CreateComplexFormComponentEntry(entry.Id, true, entry.Components, api);
+        if (createComplexForms) await CreateComplexFormComponentEntry(entry.Id, false, entry.ComplexForms, api);
+        if (createComplexFormTypes) await CreateComplexFormTypes(entry.ComplexFormTypes, api);
+        foreach (var sense in entry.Senses)
+        {
+            sense.EntryId = entry.Id;
+            foreach (var exampleSentence in sense.ExampleSentences)
+            {
+                exampleSentence.SenseId = sense.Id;
+            }
+        }
+        return entry;
+        static async Task CreateComplexFormComponentEntry(Guid entryId,
+            bool isComponent,
+            IList<ComplexFormComponent> complexFormComponents,
+            IMiniLcmApi api)
+        {
+            int i = 1;
+            foreach (var complexFormComponent in complexFormComponents)
+            {
+                //generated entries won't have the expected ids, so fix them up here
+                if (isComponent)
+                {
+                    complexFormComponent.ComplexFormEntryId = entryId;
```

Comment:
same for complex form stuff, if the id doesn't match what it should be do throw?

Reply:
I'm trying to think if there will be any references in the model where it would be logical for them to **not** be immutable. Do we expect to ever move a child object to a different parent?

Even if we find examples of where that makes sense, a general policy of "references are immutable" sounds desireable.